### PR TITLE
Add release note for 2.7.4

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation

As releasing 2.7.4, so add release note.

### Documentation
  
- [x] `no-need-doc`

### CODE REVIEW

**Review by mattisonchao — APPROVED** [review]

File: site2/website/release-notes.md [hunk 1]
```diff
@@ -797,6 +797,116 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- Cancel scheduled tasks when deleting ManagedLedgerImpl. [#12565](https://github.com/apache/pulsar/pull/12565)
+- NPE on OpAddEntry while ManagedLedger is closing. [#12364](https://github.com/apache/pulsar/pull/12364)
+- Avoid potentially blocking calls to metadata on critical threads. [#12340](https://github.com/apache/pulsar/pull/12340)
+- Fixed used after recycle issue in OpAddEntry. [#12103](https://github.com/apache/pulsar/pull/12103)
+- Optimize the memory usage of Cache Eviction. [#12045](https://github.com/apache/pulsar/pull/12045)
+- Fix wrong key-hash selector used for new consumers after all the previous consumers disconnected. [#12035](https://github.com/apache/pulsar/pull/12035)
+- Only auth_errors should log at error level. [#9325](https://github.com/apache/pulsar/pull/9325)
+
+#### Dependency upgrade
+- Upgrade Log4j2 version to 2.14.0 and replace legacy log4j dependency. [#8880](https://github.com/apache/pulsar/pull/8880)
```
Comment (codelipenghui) [inline site2/website/release-notes.md:860]: We can remove this one.

File: site2/website/release-notes.md [hunk 2]
```diff
@@ -797,6 +797,116 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- [Java] fix seek at batchIndex level receive duplicated messages. [#11826](https://github.com/apache/pulsar/pull/11826)
+- [Java] Fixed race condition on multi-topic consumer. [#11764](https://github.com/apache/pulsar/pull/11764)
+- [Java] remove consumer reference from PulsarClient on subscription failure. [#11758](https://github.com/apache/pulsar/pull/11758)
+- [Java] Fixed block forever bug in Consumer.batchReceive. [#11691](https://github.com/apache/pulsar/pull/11691)
+- [Java] Send CloseProducer on timeout. [#13161](https://github.com/apache/pulsar/pull/13161)
+- [Pulsar perf] Fix log level config for pulsar-admin, pulsar-client and pulsar-perf. [#12915](https://github.com/apache/pulsar/pull/12915)
+- [Java] Remove invalid call to Thread.currentThread().interrupt(). [#12652](https://github.com/apache/pulsar/pull/12652)
+- [C++] Removed usages of boost::regex. [#9533](https://github.com/apache/pulsar/pull/9533)
+
+#### Test
```
Comment (codelipenghui) [inline site2/website/release-notes.md:888]: We can remove this section

File: site2/website/release-notes.md [hunk 3]
```diff
@@ -797,6 +797,116 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- Fix wrong key-hash selector used for new consumers after all the previous consumers disconnected. [#12035](https://github.com/apache/pulsar/pull/12035)
+- Only auth_errors should log at error level. [#9325](https://github.com/apache/pulsar/pull/9325)
+
+#### Dependency upgrade
+- Upgrade Log4j2 version to 2.14.0 and replace legacy log4j dependency. [#8880](https://github.com/apache/pulsar/pull/8880)
+- Remove Boost::System runtime dependency. [#9498](https://github.com/apache/pulsar/pull/9498)
+- Upgrade commons-compress to 1.21. [#11345](https://github.com/apache/pulsar/pull/11345)
+- Upgrade Jetty to 9.4.43.v20210629. [#11660](https://github.com/apache/pulsar/pull/11660)
+- Add maven.restlet.org repository. [#13248](https://github.com/apache/pulsar/pull/13248)
+- Bump log4j to 2.15.0. [#13226](https://github.com/apache/pulsar/pull/13226)
```
Comment (codelipenghui) [inline site2/website/release-notes.md:865]: for 2.7.4, it should be 2.16.0?

Comment (codelipenghui) [reply site2/website/release-notes.md:865]: And it's better to create use a new section to highlight the security issue fixes.

**Comment by mattisonchao** [comment]
@Technoboy-   

maybe we should cheery pick this PR :  #13392

**Comment by Technoboy-** [comment]
> @Technoboy-
> 
> maybe we should cheery pick this PR? #13392

Yes, right.

File: site2/website/release-notes.md [hunk 4]
```diff
@@ -797,6 +797,116 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
 
 
+### 2.7.4
+
+#### 2021-12-14
+
+#### Broker
+- Fix DispatchRateLimiter does not take effect. [#8611](https://github.com/apache/pulsar/pull/8611)
+- Fix set-publish-rate when using preciseTopicPublishRateLimiterEnable=true. [#10384](https://github.com/apache/pulsar/pull/10384)
+- fix retention policy in topic policy not work. [#11021](https://github.com/apache/pulsar/pull/11021)
```
Comment (Anonymitaet) [inline site2/website/release-notes.md:807]: Please keep consistent for the following aspects: 
- verb tense (use **present tense**). reason: https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.e8uqh1awkcnp
- capitalization 
    - use sentence-style capitalization. details: https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.s1g5tosy5w3i
    - capitalize proper nouns correctly. details: https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.je5thtkj42cb

Besides, please pay attention to the following aspects：
- do not use contractions in technical information. reason: https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.qa5rwiehb245


Can you check all occurrences overall and update them? Thanks

File: site2/website/release-notes.md [hunk 5]
```diff
@@ -797,6 +797,116 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- [Java] Fix Consumer listener does not respect receiver queue size. [#11455](https://github.com/apache/pulsar/pull/11455)
+- [Java] Add a optional params scope for pulsar oauth2 client. [#11931](https://github.com/apache/pulsar/pull/11931)
+- [Java] fix seek at batchIndex level receive duplicated messages. [#11826](https://github.com/apache/pulsar/pull/11826)
+- [Java] Fixed race condition on multi-topic consumer. [#11764](https://github.com/apache/pulsar/pull/11764)
+- [Java] remove consumer reference from PulsarClient on subscription failure. [#11758](https://github.com/apache/pulsar/pull/11758)
+- [Java] Fixed block forever bug in Consumer.batchReceive. [#11691](https://github.com/apache/pulsar/pull/11691)
+- [Java] Send CloseProducer on timeout. [#13161](https://github.com/apache/pulsar/pull/13161)
+- [Pulsar perf] Fix log level config for pulsar-admin, pulsar-client and pulsar-perf. [#12915](https://github.com/apache/pulsar/pull/12915)
+- [Java] Remove invalid call to Thread.currentThread().interrupt(). [#12652](https://github.com/apache/pulsar/pull/12652)
+- [C++] Removed usages of boost::regex. [#9533](https://github.com/apache/pulsar/pull/9533)
```
Comment (Anonymitaet) [inline site2/website/release-notes.md:886]: can you get them organized to improve readability? something like:

```
- [Java] xx
- [Java] xx
...
- [C++] xx
- [C++] xx
...
- [Python] xx
- [Python] xx
...
```

**Review by Anonymitaet — CHANGES_REQUESTED** [review]

**Comment by Anonymitaet** [comment]
@Technoboy- can we re-organize the structure to the following? So that users can capture the desired info in a quicker manner

```

2.x.x release notes

- Enhancement
  - Security
  - Broker
  - xxx

- Bug fix
  - Broker
  - Client
  - xxx

- Miscellaneous 
  - xxx
  - xxx
  - xxx
```

File: site2/website/release-notes.md (hunk 1 above)
Comment (Technoboy-) [reply site2/website/release-notes.md:860]: done

File: site2/website/release-notes.md (hunk 2 above)
Comment (Technoboy-) [reply site2/website/release-notes.md:888]: ok

File: site2/website/release-notes.md (hunk 4 above)
Comment (Technoboy-) [reply site2/website/release-notes.md:807]: All right.

File: site2/website/release-notes.md (hunk 5 above)
Comment (Technoboy-) [reply site2/website/release-notes.md:886]: Ok

File: site2/website/release-notes.md (hunk 3 above)
Comment (Technoboy-) [reply site2/website/release-notes.md:865]: Ok.

**Review by eolivelli — APPROVED** [review]
LGTM

File: site2/website/release-notes.md [hunk 6]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- Fix the topic in fenced state and can not recover. [#11737](https://github.com/apache/pulsar/pull/11737)
+- Support disabling non-TLS service ports. [#11724](https://github.com/apache/pulsar/pull/11724)
+- Fix the bug, can not update topic when the update topicName is contained by an existed topic as a
+  part. [#11686](https://github.com/apache/pulsar/pull/11686)
+- Fix Pulsar didn't respond error messages when throw
+  InterceptException. [#11650](https://github.com/apache/pulsar/pull/11650)
+- Avoid redundant calls for getting the offload policies from the
+  offloader. [#11629](https://github.com/apache/pulsar/pull/11629)
+- Optimize ManagedLedger Ledger Ownership Check. [#13222](https://github.com/apache/pulsar/pull/13222)
+- Do not reuse the Failed OpAddEntry object which lead bundle unloading
```
Comment (Anonymitaet) [inline site2/website/release-notes.md:865]:
```suggestion 
- Do not reuse the Failed OpAddEntry object which leads to bundle unloading
```

File: site2/website/release-notes.md [hunk 7]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- Optimize the memory usage of Cache Eviction. [#12045](https://github.com/apache/pulsar/pull/12045)
+- Fix wrong key-hash selector used for new consumers after all the previous consumers
+  disconnected. [#12035](https://github.com/apache/pulsar/pull/12035)
+- Only auth_errors should log at error level. [#9325](https://github.com/apache/pulsar/pull/9325)
+
+#### Dependency upgrade
+- Remove Boost::System runtime dependency. [#9498](https://github.com/apache/pulsar/pull/9498)
+- Upgrade Jetty to 9.4.43.v20210629. [#11660](https://github.com/apache/pulsar/pull/11660)
+- Add maven.restlet.org repository. [#13248](https://github.com/apache/pulsar/pull/13248)
+- Upgrade Zookeeper version to 3.5.9. [#12981](https://github.com/apache/pulsar/pull/12981)
```
Comment (Anonymitaet) [inline site2/website/release-notes.md:887]:
```suggestion
- Upgrade ZooKeeper version to 3.5.9. [#12981](https://github.com/apache/pulsar/pull/12981)
```

File: site2/website/release-notes.md [hunk 8]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+#### Client
+
+- [Java] Refactor seek to reuse common logic. [#9670](https://github.com/apache/pulsar/pull/9120)
+- [Java] Process partitioned-topic messages on different
+  listener-threads. [#10017](https://github.com/apache/pulsar/pull/10017)
+- [Java] Fix Consumer listener does not respect receiver queue
+  size. [#11455](https://github.com/apache/pulsar/pull/11455)
+- [Java] Add a optional params scope for pulsar oauth2 client. [#11931](https://github.com/apache/pulsar/pull/11931)
+- [Java] Fix seek at batchIndex level receive duplicated messages. [#11826](https://github.com/apache/pulsar/pull/11826)
+- [Java] Fixed race condition on multi-topic consumer. [#11764](https://github.com/apache/pulsar/pull/11764)
```
Comment (Anonymitaet) [inline site2/website/release-notes.md:901]:
```suggestion
- [Java] Fix race condition on multi-topic consumer. [#11764](https://github.com/apache/pulsar/pull/11764)
```

File: site2/website/release-notes.md [hunk 9]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- [Java] Remove consumer reference from PulsarClient on subscription
+  failure. [#11758](https://github.com/apache/pulsar/pull/11758)
+- [Java] Fixed block forever bug in Consumer.batchReceive. [#11691](https://github.com/apache/pulsar/pull/11691)
```
Comment (Anonymitaet) [inline site2/website/release-notes.md:904]:
```suggestion
- [Java] Fixeblock forever bug in Consumer.batchReceive. [#11691](https://github.com/apache/pulsar/pull/11691)
```

File: site2/website/release-notes.md [hunk 10]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- [Java] Send CloseProducer on timeout. [#13161](https://github.com/apache/pulsar/pull/13161)
+- [Java] Remove invalid call to Thread.currentThread().interrupt()
+  . [#12652](https://github.com/apache/pulsar/pull/12652)
+- [Python] Fix fields that are ignoring the required key argument. [#11508](https://github.com/apache/pulsar/pull/11508)
+- [C++] Removed usages of boost::regex. [#9533](https://github.com/apache/pulsar/pull/9533)
```
Comment (Anonymitaet) [inline site2/website/release-notes.md:909]:
```suggestion
- [C++] Remove usages of boost::regex. [#9533](https://github.com/apache/pulsar/pull/9533)
```

File: site2/website/release-notes.md [hunk 11]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- [C++] Use sendRequestWithId to add timeout to
+  hasMessageAvailable. [#11600](https://github.com/apache/pulsar/pull/11600)
+- [Pulsar perf] Fix log level config for pulsar-admin, pulsar-client and
+  pulsar-perf. [#12915](https://github.com/apache/pulsar/pull/12915)
+
+#### Functions and Pulsar IO
+
+- Fix source stats exposing empty exceptions list. [#11478](https://github.com/apache/pulsar/pull/11478)
+- Fixes function api can not use authdata to check superuser. [#11418](https://github.com/apache/pulsar/pull/11418)
```
Comment (Anonymitaet) [inline site2/website/release-notes.md:918]:
```suggestion
- Fix function API can not use authdata to check superuser. [#11418](https://github.com/apache/pulsar/pull/11418)
```

**Review by codelipenghui — APPROVED** [review]

File: site2/website/release-notes.md [hunk 12]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
 - [C++] Remove namespace check for MultiTopicsConsumerImpl [#9520](https://github.com/apache/pulsar/pull/9520)
 
 
+### 2.7.4
+
+#### 2021-12-14
+
+#### Security
+
+- Upgrade log4j to 2.17.0 - [CVE-2021-45105](https://pulsar.apache.org/blog/2021/12/11/Log4j-CVE/)
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply site2/website/release-notes.md:806]: done

File: site2/website/release-notes.md [hunk 13]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+
+- Fix DispatchRateLimiter does not take effect. [#8611](https://github.com/apache/pulsar/pull/8611)
+- Fix set-publish-rate when using
+  preciseTopicPublishRateLimiterEnable=true. [#10384](https://github.com/apache/pulsar/pull/10384)
+- Fix retention policy in topic policy not work. [#11021](https://github.com/apache/pulsar/pull/11021)
+- Fix GetTopicsOfNamespace with binary lookup service not check
+  auth. [#11172](https://github.com/apache/pulsar/pull/11172)
+- Fix retention of keys in compaction. [#11287](https://github.com/apache/pulsar/pull/11287)
+- Fix using partitioned topic name to get Policy. [#11294](https://github.com/apache/pulsar/pull/11294)
+- Fix npe when unloading persistent partitioned topic. [#11310](https://github.com/apache/pulsar/pull/11310)
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply site2/website/release-notes.md:821]: done

File: site2/website/release-notes.md [hunk 14]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- Close the replicator and replication client when delete cluster. [#11342](https://github.com/apache/pulsar/pull/11342)
+- Remove RateLimiter constructors and replace with builder usage. [#11599](https://github.com/apache/pulsar/pull/11599)
+- Add backoff for setting for getting topic policies. [#11574](https://github.com/apache/pulsar/pull/11574)
+- Fix decode compression managedLedgerInfo data. [#11569](https://github.com/apache/pulsar/pull/11569)
+- Fix some topic policy operation without backoff. [#11560](https://github.com/apache/pulsar/pull/11560)
+- Fix data lost when using earliest position to subscribe to a
+  topic. [#11547](https://github.com/apache/pulsar/pull/11547)
+- Fix the schema deletion when delete topic with delete schema. [#11501](https://github.com/apache/pulsar/pull/11501)
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply site2/website/release-notes.md:829]: done

File: site2/website/release-notes.md [hunk 15]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- Do not create system topic for heartbeat namespace. [#11499](https://github.com/apache/pulsar/pull/11499)
+- Compress managed ledger info. [#11490](https://github.com/apache/pulsar/pull/11490)
+- Add backoff for setting for getting topic policies. [#11487](https://github.com/apache/pulsar/pull/11487)
+- Solving precise rate limiting does not takes effect. [#11446](https://github.com/apache/pulsar/pull/11446)
+- Fix replicated subscriptions direct memory leak. [#11396](https://github.com/apache/pulsar/pull/11396)
+- Close the replicator and replication client when delete cluster. [#11390](https://github.com/apache/pulsar/pull/11390)
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply site2/website/release-notes.md:835]: done

File: site2/website/release-notes.md (hunk 8 above)
Comment (Technoboy-) [reply site2/website/release-notes.md:901]: done

File: site2/website/release-notes.md (hunk 10 above)
Comment (Technoboy-) [reply site2/website/release-notes.md:909]: done

File: site2/website/release-notes.md (hunk 11 above)
Comment (Technoboy-) [reply site2/website/release-notes.md:918]: done

File: site2/website/release-notes.md [hunk 16]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- Add backoff for setting for getting topic policies. [#11574](https://github.com/apache/pulsar/pull/11574)
+- Fix decode compression managedLedgerInfo data. [#11569](https://github.com/apache/pulsar/pull/11569)
+- Fix some topic policy operation without backoff. [#11560](https://github.com/apache/pulsar/pull/11560)
+- Fix data lost when using earliest position to subscribe to a
+  topic. [#11547](https://github.com/apache/pulsar/pull/11547)
+- Fix the schema deletion when delete topic with delete schema. [#11501](https://github.com/apache/pulsar/pull/11501)
+- Do not create system topic for heartbeat namespace. [#11499](https://github.com/apache/pulsar/pull/11499)
+- Compress managed ledger info. [#11490](https://github.com/apache/pulsar/pull/11490)
+- Add backoff for setting for getting topic policies. [#11487](https://github.com/apache/pulsar/pull/11487)
+- Solving precise rate limiting does not takes effect. [#11446](https://github.com/apache/pulsar/pull/11446)
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply site2/website/release-notes.md:833]: done

File: site2/website/release-notes.md (hunk 7 above)
Comment (Technoboy-) [reply site2/website/release-notes.md:887]: done

**Review by Anonymitaet — APPROVED** [review]

File: site2/website/release-notes.md [hunk 17]
```diff
@@ -797,6 +797,140 @@ For more context about this issue, see [[Pulsar Functions and IO] Cannot upgrade
...
+- Fix replicated subscriptions direct memory leak. [#11396](https://github.com/apache/pulsar/pull/11396)
+- Close the replicator and replication client when delete cluster. [#11390](https://github.com/apache/pulsar/pull/11390)
+- Invalidate the read handle after all cursors consumed. [#11389](https://github.com/apache/pulsar/pull/11389)
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply site2/website/release-notes.md]: Done

**Pull request merged**